docker-images/kafka-based/kafka/scripts/kafka_run.sh

```diff
@@ -63,6 +63,12 @@ if [ "$STRIMZI_KRAFT_ENABLED" = "true" ]; then
     echo "Kraft storage is already formatted"
   fi
 
+  # remove quorum-state file so that we won't enter voter not match error after scaling up/down
+  if [ -f "$KRAFT_LOG_DIR/__cluster_metadata-0/quorum-state" ]; then
+    echo "Removing quorum-state file"
+    rm -f "$KRAFT_LOG_DIR/__cluster_metadata-0/quorum-state"
+  fi
```

Comment:
is it safe to remove quorum-state file even just when brokers are rolled for any other reason? (i.e. config change, updates, ...) So I mean, when it's not about scale up/down ...

Reply:
I know what you mean. This is safe. The `quorum-state` file contains the voter's info and the leader's info. That means, Before this change, when a pod is restarted, the controller (suppose it is not the leader) can directly fetch from the quorum leader node. After this change, we have to find the current quorum leader each time the pod is restarted. I think that's the only difference. 


I think it should be fine. WDYT?

Reply:
More rolling happens in specific scenarios (i.e. certificates renewal, ...) but it should not happen that frequently.
I guess this "fine the current quorum leader" is about some RPC calls.
I can live with it. @scholzj wdyt?

Reply:
> I guess this "fine the current quorum leader" is about some RPC calls.

Correct